source/darwin/secure_transport_tls_channel_handler.c

```diff
@@ -540,7 +540,11 @@ static int s_process_write_message(
 
     if (status != noErr) {
         AWS_LOGF_DEBUG(
-            AWS_LS_IO_TLS, "id=%p: SSLWrite failed with OSStatus error code %d.", (void *)handler, (int)status);
+            AWS_LS_IO_TLS,
+            "id=%p: SSLWrite failed with OSStatus error code %d (%s).",
+            (void *)handler,
+            (int)status,
+            CFStringGetCStringPtr(SecCopyErrorMessageString(status, NULL), kCFStringEncodingUTF8));
```

Comment:
Is this the only place where we need to decode an error in the secure transport code? Feels like there might be more?

Reply:
Also, here's some complication you need to  worry about

From the [SecCopyErrorMessageString docs](https://developer.apple.com/documentation/security/1394686-seccopyerrormessagestring?language=objc):
> returns NULL if no string is available

and also

> Call the CFRelease function to release this object when you are finished using it

And beware what you pass to [CFRelease](https://developer.apple.com/documentation/corefoundation/1521153-cfrelease?language=objc):
> This value must not be NULL. 

oh yeah, [CFStringGetCStringPtr](https://developer.apple.com/documentation/corefoundation/1542133-cfstringgetcstringptr?language=objc) can fail too:
> Whether or not this function returns a valid pointer or NULL depends on many factors, all of which depend on how the string was created and its properties
so maybe we should use [CFStringGetCString](https://developer.apple.com/documentation/corefoundation/1542721-cfstringgetcstring?language=objc) instead which copies into a buffer, but won't randomly fail.

so you'll probably need to do a bunch of horrible error checking like
```
CFStringRef msg_cfstring = SecCopyMessageString(status, NULL);
const char msg_buf[512];
bool msg_buf_valid = false;
if (msg_cfstring) {
    msg_buf_valid = CFStringGetCString(msg_cfstring, msg_buf, sizeof(msg_buf), kCFStringEncodingUTF8);
}
if (!msg_buf_valid) {
   strncpy(msg_buf, "Unknown", sizeof(msg_buf));
}

AWS_LOGF_DEBUG("blah blah %s",...., msg_buf);

if (msg_cfstring) {
    CFRelese(msg_cfstring);
}


```